### ISSUE TITLE
/muteUniverse on dyn universe channels

### ISSUE THREAD

**Issue by ryzom-pipeline** [ryzom/ryzomcore#143]
**[Original report](https://bitbucket.org/ryzom/ryzomcore/issue/143) by Meelis Mägi (Bitbucket: [Meelis Mägi](https://bitbucket.org/Meelis Mägi), ).**

----------------------------------------

I believe there is a bug in /muteUniverse command.

```
#!c++

--- a/code/ryzom/server/src/input_output_service/chat_manager.cpp
+++ b/code/ryzom/server/src/input_output_service/chat_manager.cpp
@@ -639,9 +639,9 @@
                                        // If universal channel check if player muted
                                        if (session->getChan()->UniversalChannel)
                                        {
-                                               if(_MutedUsers.find( eid ) != _MutedUsers.end())
+                                               if(_MutedUniverseUsers.find( eid ) != _MutedUniverseUsers.end())
                                                {
-                                                       nldebug("IOSCM:  chat The player %s:%x is muted",
+                                                       nldebug("IOSCM:  chat The player %s:%x is universe muted",
                                                                TheDataset.getEntityId(sender).toString().c_str(),
                                                                sender.getIndex());
                                                        return;

```

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

What does this change?

**Comment by ryzom-pipeline** [comment]
**Original comment by Meelis Mägi (Bitbucket: [Meelis Mägi](https://bitbucket.org/Meelis Mägi), ).**

----------------------------------------

That change will probably prevent players that are muted with /muteUniverse command, also talk to dyn channels marked as universal (faction, lang)... probably.

If you check the function, then _MutedUsers is [already checked at start](https://bitbucket.org/ryzom/ryzomcore/src/5d49a05bf2acb51fdad694ebca092094690ff9c3/code/ryzom/server/src/input_output_service/chat_manager.cpp?at=default#cl-480) and [ChatGroup::universe](https://bitbucket.org/ryzom/ryzomcore/src/5d49a05bf2acb51fdad694ebca092094690ff9c3/code/ryzom/server/src/input_output_service/chat_manager.cpp?at=default#cl-576) is checking _MutedUniverseUsers, so my only conclusion it that there is a typo.

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

Yup, okay, looks like it.

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

Fix #143 /muteUniverse on dyn universe channels
